### PR TITLE
Fix strerror handling

### PR DESCRIPTION
**Pull request by jandrej**
Resolves #2995

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2995](https://github.com/mfem/mfem/pull/2995) | @jandrej | @tzanio | @tzanio | 5/4/22 | 5/8/22 | 5/8/22 | |

# PR Checklist
    
- [x] Code builds.
- [x] Code passes `make style`.

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

File: general/zstr.hpp [hunk 1]
```diff
@@ -59,27 +59,28 @@
...
-   }
-#elif (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE || \
-      defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
-      defined(__NetBSD__) || defined(__DragonFly__) || defined(__EMSCRIPTEN__)
-   // XSI-compliant strerror_r()
-   if (strerror_r(errno, &buff[0], buff.size()) != 0)
-   {
-      buff = "Unknown error";
-   }
+   buff = "unknown error";
```
Comment (tzanio) [inline general/zstr.hpp:80]: Why change the Windows logic?

Comment (jandrej) [reply general/zstr.hpp:80]: that's a mistake from my revert

**Comment by tzanio** [comment]
Do we have anything that will actually test that portion of the code?

**Comment by jandrej** [comment]
> Do we have anything that will actually test that portion of the code?

The funny thing is, after the transition to an hpp only file, we never use the strerror() method :)

**Comment by tzanio** [comment]
@reine249, can you please review?

**Comment by tzanio** [comment]
I see a new version in https://github.com/mateidavid/zstr, including https://github.com/mateidavid/zstr/blob/master/src/strict_fstream.hpp#L67.

I don't think we should upgrade if we don't _need_ to though.

**Review by reine249 — APPROVED** [review]
Thanks again for the quick fix.

**Review by tzanio — APPROVED** [review]

**Review by tomstitt — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**